store.go

```diff
@@ -2934,11 +2978,35 @@ func (s *store) Mounted(id string) (int, error) {
 }
 
 func (s *store) UnmountImage(id string, force bool) (bool, error) {
+	if err := s.startUsingGraphDriver(); err != nil {
+		return false, err
+	}
+	defer s.stopUsingGraphDriver()
+
+	if err := s.imageStore.startReading(); err != nil {
+		return false, err
+	}
+	defer s.imageStore.stopReading()
+
 	img, err := s.Image(id)
 	if err != nil {
 		return false, err
 	}
-	return s.Unmount(img.TopLayer, force)
+
+	return writeToLayerStore(s, func(rlstore rwLayerStore) (bool, error) {
+		// check if the layer with the canonical mapping is in the mapped top layers
+		for _, layerID := range img.MappedTopLayers {
+			l, err := rlstore.Get(layerID)
+			if err != nil {
+				return false, err
```

Comment:
The previous version silently ignored `ErrLayerUnknown`. Is the change intentional?

I don’t have much of an opinion, failing immediately could be argued to be better, I just want to avoid accidents.

Reply:
before we were running without keeping the lock, so that error was expected.  Now, I am not sure we should ignore it.

Reply:
Thanks, good point.

We are back to loading the image, unlocking, and then looking at layers, so we might again hit a race here. Should this go back to the previous behavior?